src/units/unit.cpp

```diff
@@ -2214,7 +2214,10 @@ void unit::apply_builtin_effect(std::string apply_to, const config& effect)
 		}
 
 		if(increase.empty() == false) {
-			experience_ = utils::apply_modifier(experience_, increase, 1);
```

Comment:
The third argument is a minimum value; setting it to zero fixes the bug.

@CelticMinstrel can you remember why it was set to 1 in #618?
Having a minimum of 1 for max_experience makes sense to me, so maybe it was copied from that.

Reply:
I don't remember, no.

Reply:
> The third argument is a minimum value; setting it to zero fixes the bug.
> 
fixed